### ISSUE TITLE
Communication btw boxes and server(s)

### ISSUE THREAD

**Issue by clemsos** [ludobox/ludobox#13]
A basic implementation of a straightforward process to download game data from 3rd parts using 2 different protocols : 

* **http request** : crawling the official website (http://ludobox.net) or any other ludobox to obtain games over a http network
* **dat** : sync data from any devices using the [Dat protocol](https://github.com/datproject/dat), allowing to easily build a larger mesh/P2P network (including the official webserver)

At this stage, it looks important to maintain a general index file on the central server containing basic info about all games like names, dat ID and other relevant items.

**Comment by clemsos** [comment]
This has been vastly improved by #17 and the new API architecture - still some work though.  

Some notes about the current specs : 

- lets keep **dat** for later and focus on a dingle, simple solution - but dat is awesome still
- all backends are JSON APIs, accessible from anywhere by any client (all CORS)
- clients are used to send instructions and coordinate different backend (ex : "box at ip XX.XX.XX, please download the game at this address ip XX.XX.XX") 
- download of a remote game is started by a posting (as in POST request) the description of what should be downloaded at the `/api/clone` route.

So, basically : 
1. your client download a list of all games available on a remote server
2. use select one of the game on the client 
3. client download a JSON description (ajax) containing :  game full description + URL  + list of atachements (files)
4. client send these instructions to its local server
5. local server download all files from remote server
6. local server ping client when all downloads are done

This can surely be improved but that is the state of my thought right now
